src/NumericalAlgorithms/LinearOperators/PartialDerivatives.tpp

```diff
@@ -7,25 +7,214 @@
 
 #include "DataStructures/DataBox/DataBoxTag.hpp"
 #include "DataStructures/DataBox/Prefixes.hpp"
+#include "DataStructures/DataVector.hpp"
 #include "DataStructures/Matrix.hpp"
 #include "DataStructures/Variables.hpp"
 #include "Domain/Mesh.hpp"
 #include "NumericalAlgorithms/LinearOperators/Transpose.hpp"
 #include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "Utilities/Algorithm.hpp"
 #include "Utilities/Blas.hpp"
+#include "Utilities/Gsl.hpp"
 #include "Utilities/MakeArray.hpp"
 #include "Utilities/StdArrayHelpers.hpp"
 
 namespace partial_derivatives_detail {
 template <size_t Dim, typename VariableTags, typename DerivativeTags>
 struct LogicalImpl;
+
+// This routine has been optimized to perform really well. The following
+// describes what optimizations were made.
+//
+// - The `partial_derivatives` functions below have an overload where the
+//   logical derivatives may be passed in instead of being computed. In the
+//   overloads where the logical derivatives are not passed they must be
+//   computed. However, it is more efficient to allocate the memory for the
```

Comment:
"must be computed" -> "must be allocated" ?

Reply:
Nope, must be computed

Reply:
Okay 👍